### ISSUE TITLE
Unable to Run Debug Code Due To No Qt Platform Plugin

### ISSUE THREAD

**Issue by wagrenier** [jpd002/Play-#1022]
Using both Qt Creator or CLion on Windows, I get the error `qt.qpa.plugin: Could not find the Qt platform plugin "windows" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.` when trying to run the application. I have no `QT_PLUGIN` set in my env var, which usually is the problem. It builds correctly, but fails when creating the application in the main. I am using `Qt 5.15.2` with `msvc2019_64`. I'll continue to try to fix the issue and update the issue in case I find a solution.

**Comment by Zer0xFF** [comment]
I've noticed this issue since returning to use windows as my regular OS.

for now, i copy the "plugins" folder from C:\Qt\Qt5.12.9\5.12.9\msvc2017_64 and place it beside the exe.

I think we can get around this issue by setting up a post build cmd within cmake to trigger `windeployqt` after build, though im on the fence about it, as it will trigger everytime the exe is rebuilt

**Comment by wagrenier** [comment]
Could it be made during the initial Build of the project, while generating files from the main `CMakeLists.txt`. Instead of being executed every Build.

**Comment by Zer0xFF** [comment]
the thing is, `windeployqt` copies only the needed dlls by analysing the executable, if the exe doesnt exist, it cant analyse it. unless we copy all the plugins.

**Comment by wagrenier** [comment]
Right now I copied the entire plugins folder into the Build output directory and it works well. However, it does increase the output build size locally.

**Comment by Zer0xFF** [comment]
I dont think local build size is a concern, since the entire `plugins` folder is only 66MB, which is dwarfed by the size of the entire build directory of 3.3GB. if you're building to install, then running `windeployqt` isnt an issue since its a one time thing.

though something like this does the trick https://github.com/Zer0xFF/Play-/commit/64e43322dab59ac03295ce4c4386a99faa6c2b12
```
real    0m0.076s
user    0m0.015s
sys     0m0.000s
```
time added is negligable as well

**Comment by wagrenier** [comment]
I can try to add a command to copy the folder as mentioned. That way, for future people it could be simpler to get into the project. In the next week I'll do a PR to improve the setup documentation for Windows with VS2019 and CLion.

On another topic, any way that I can get started up on this project? I wanna help during my free time, but I can't find any issues with Good First Issue or a board with what is priority or needed.

Thanks!

**Comment by Zer0xFF** [comment]
> On another topic, any way that I can get started up on this project? I wanna help during my free time, but I can't find any issues with Good First Issue or a board with what is priority or needed.

lets see, that really depends on your knowledge base and perhaps what you're interested in, though as an ice breaker, i think the screenshot feature can use a small improvement, as is now the screenshot feature captures images as BMP which is not entirely useful, but can also produce ridiculously large images, especially on higher resoutions. so converting that to png would be useful. https://github.com/jpd002/Play-/blob/b8d9fc763d3d46ac45809eff96df9209781f29d5/Source/ScreenShotUtils.cpp#L19

I think `stb_image.h` https://github.com/nothings/stb or libpng might work here http://www.libpng.org/pub/png/libpng.html

**Comment by jpd002** [comment]
Started getting this problem as I updated my local Qt version to 5.15.2. I tried the `windeployqt` trick post build, seems to be working good: https://github.com/jpd002/Play-/commit/06e2d60474aaef7f2ffbf9d8287f92f1a01d1122. The tool seems to be smart enough to not copy the files if they are already there.

Loosely inspired from what Dolphin does: https://github.com/dolphin-emu/dolphin/blob/6086111566f26bbe298daa4e477ef690d18c9e80/Source/Core/DolphinQt/CMakeLists.txt#L390.
